example/helloWorld/src/helloworld.cpp

```diff
@@ -0,0 +1,190 @@
+						     threadsExtent );
+    return globalThreadIdx[0];
+}
+
+/**
+ * Hello World Kernel
+ *
+ * Prints "[tid] Hello World" where tid is
+ * the global thread number.
+ */
+struct HelloWorldKernel {
+    template <typename T_Acc>
+    ALPAKA_FN_ACC void operator()( T_Acc const &acc) const {
+        printf("[%u] Hello World\n", static_cast<unsigned>(globalThreadIdx(acc)));
+
+    }
+
+};
+
+int main() {
+
+    /**
+     * Define accelerator types
+     *
+     * It is possible to choose from a set of accelerators
+     * that are defined in the alpaka::acc namespace e.g.:
+     * - AccGpuCudaRt
+     * - AccCpuThreads
+     * - AccCpuFibers
+     * - AccCpuOmp2Threads
+     * - AccCpuOmp2Blocks
+     * - AccCpuOmp4
+     * - AccCpuSerial
+     *
+     * Each accelerator has strengths and weaknesses. Therefore,
+     * they need to be choosen carefully depending on the actual
+     * use case. Furthermore, some accelerators only support a
+     * particular workdiv, but workdiv can also be generated
+     * automatically.
+     *
+     */
+    using Dim     = alpaka::dim::DimInt<3>;
+    using Size    = std::size_t;
+    using Host    = alpaka::acc::AccCpuSerial<Dim, Size>;
+    using Acc     = alpaka::acc::AccCpuSerial<Dim, Size>;
+    using Stream  = alpaka::stream::StreamCpuSync;
+    using DevAcc  = alpaka::dev::Dev<Acc>;
+    using DevHost = alpaka::dev::Dev<Host>;
+    using WorkDiv = alpaka::workdiv::WorkDivMembers<Dim, Size>;
+
+
+    /**
+     * Get the first devices
+     *
+     * The accelerator only defines how something should be
+     * parallized, but a device is the real entity which will
+     * run the parallel programm. The device can be choosen
+     * by id (0 to the number of devices minus 1) or you
+     * can also retrieve all devices in a vector (getDevs()).
+     * In this example the first devices is choosen.
+     *
+     */
+    DevAcc  devAcc  (alpaka::dev::DevMan<Acc>::getDevByIdx(0));
+    DevHost devHost (alpaka::dev::DevMan<Host>::getDevByIdx(0));
+
+    
+    /**
+     * Create a stream to the accelerator device
+     *
+     * A stream can be interpreted as the work queue
+     * of a particular device. Streams are filled with
+     * execution units and alpaka takes care that these
+     * units will be executed. Streams are provided in
+     * async and sync variants. 
+     * The example stream is a sync stream to a cpu device, 
+     * but it also exists an async stream for this 
+     * device (StreamCpuAsync).
+     */
+    Stream  stream  (devAcc);
+
+
+    /**
+     * Init workdiv
+     *
+     * The workdiv defines how many instances of a
+     * kernel can be spawned in parallel. How many kernels
+     * are really spawned finally depends on the choosen
+     * accelerator e.g. the sequential accelerator will
+     * only spawn one kernel at a time, whereas a CUDA
+     * accelerator can spawn hundreds of kernels at the
+     * same time.
```

Comment:
I have reworded it a bit:
A kernel is executed for each element of a n-dimensional grid distinguished by the element indices.
The work division defines the number of instantiations as well as the type of parallelism used.
Different accelerators have different requirements on the work division. For example, the seuential accelerator can not provide any thread level parallelism (synchronizable as well as non synchronizable), whereas the CUDA accelerator can spawn hundreds of synchronizing and non synchronizing threads at the same time.